### PR TITLE
Stepping over wfi does not enter wait state.

### PR DESCRIPTION
**Pull request by timsifive**

### CODE REVIEW

**Comment by mwachs5** [comment]
Does the spec clearly state somewhere that debug halt requests force wfi to compete?

**Comment by timsifive** [comment]
I agree it's a bit convoluted. I'm accounting for the possibility that wfi causes an exception. From the privspec:
> This instruction may raise an illegal instruction exception when TW=1 in mstatus, as described in Section 3.1.10.

In that case the exception must happen, and it should not be a nop.

The spec doesn't say that wfi is interrupted by halt. I'll add that.

**Comment by mwachs5** [comment]
Your reading of the spec is closer than mine, sounds good!

File: core_debug.tex [hunk 1]
```diff
@@ -66,6 +66,12 @@ \section{Load-Reserved/Store-Conditional Instructions}
     automate this.
 \end{commentary}
 
+\section{Wait for Interrupt Instruction}
+
+If halt is requested while {\tt wfi} is executing, then the hart must leave the
+stalled state, completing this instruction's execution, and then enter Debug
```
Comment (mwachs5) [inline core_debug.tex]: is there a reason to use "stalled' here and "wait" in your other change? Aren't they the same state? Can we use the same term?

Comment (timsifive) [reply core_debug.tex]: I'll make it consistently use stalled (which is the term used in the priv spec).

**Pull request merged**